core/templates/dev/head/pages/email_dashboard/EmailDashboardServices.js

```diff
@@ -0,0 +1,120 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Services for oppia email dashboard page.
+ */
+
+oppia.factory('EmailDashboardDataService', [
+  '$http', '$timeout', function($http, $timeout) {
+    var QUERY_DATA_URL = '/emaildashboarddatahandler';
+    var QUERY_STATUS_CHECK_URL = '/querystatuscheck';
+    // No. of query results to display on a single page.
+    var QUERIES_PER_PAGE = 10;
+    // Store latest cursor value for fetching next query page.
+    var latestCursor = null;
+    // Array containing all fetched queries.
+    var queries = [];
+    // Index of currently-shown page of query results.
+    var currentPageIndex = -1;
+
+    var fetchQueriesPage = function(pageSize, cursor) {
+      return $http.get(QUERY_DATA_URL, {
+        params: {
+          num_of_queries_per_page: pageSize,
+          cursor: cursor
+        }
+      }).then(function(response) {
+        return response.data;
+      });
+    };
+
+    return {
+      getQueries: function() {
+        return queries;
+      },
+
+      getCurrentPageIndex: function() {
+        return currentPageIndex;
+      },
+
+      getLatestCursor: function() {
+        return latestCursor;
+      },
+
+      submitQuery: function(data, callback) {
+        var startQueryIndex = (currentPageIndex) * QUERIES_PER_PAGE;
+        var endQueryIndex = (currentPageIndex + 1) * QUERIES_PER_PAGE;
+
+        $http.post(QUERY_DATA_URL, {
+          data: data
+        }).success(function() {
+          // Wait for 500ms and then fetch details of newly submitted query.
```

Comment:
Why don't you just get the POST to return this data, instead of doing an additional fetch?